pennylane/qchem/openfermion_obs.py

```diff
@@ -837,6 +839,13 @@ def molecular_hamiltonian(
 
     |
 
+    .. warning::
```

Comment:
Do we need a deprecation warning saying that `molecular_hamiltonian` will return a `Sum` object instead of `Hamiltonian`?

Reply:
Nope, this behaviour will only happen if `active_new_opmath()` is `True`. Otherwise it will still return a `Hamiltonian`